src/queue.c

```diff
@@ -6264,8 +6267,15 @@ _dispatch_worker_thread(void *context)
 static unsigned WINAPI
 _dispatch_worker_thread_thunk(LPVOID lpParameter)
 {
-  _dispatch_worker_thread(lpParameter);
-  return 0;
+	HRESULT hr = RoInitialize(RO_INIT_MULTITHREADED);
```

Comment:
Why `RoInitialize`?  I think that limiting this to `CoInitialize` is better.  I am still concerned about the threading model though - MTA cannot make input-synchronized calls, and this will convert async to sync due to the MTA, plus this disables the message filtering (or is this equivalent to `COINIT_MULTITHREADED` and this marks everything for free threading?).  I'm not sure if we really want to use the `RoInitialize` here rather than `CoInitialize`.  The COM initializer gives greater flexibility (e.g. disabling DDE, trading off memory for speed, etc).

Reply:
Thanks for your feedback! We actually tried both `CoInitialize` and `RoInitialize`, and both seemed to work in our tests. I couldn’t find any detailed description on about the differences, and we only chose the latter because of the documentation recommending it for use with Windows Runtime APIs.

I updated it with `COINIT_MULTITHREADED | COINIT_DISABLE_OLE1DDE`, does that make sense in this context?